presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -464,6 +466,40 @@ private DataTypeTest jsonTestCases(DataType<String> jsonDataType)
                 .addRoundTrip(jsonDataType, "[]");
     }
 
+    @Test
+    public void testPrestoCreatedFloatingPoint()
+    {
+        singlePrecisionFloatingPointTests(realDataType())
+                .execute(getQueryRunner(), prestoCreateAsSelect("presto_test_float"));
+        doublePrecisionFloatinPointTests(doubleDataType())
+                .execute(getQueryRunner(), prestoCreateAsSelect("presto_test_double"));
+    }
+
+    @Test
+    public void testMysqlCreatedFloatingPoint()
+    {
+        singlePrecisionFloatingPointTests(mysqlFloatDataType())
+                .execute(getQueryRunner(), mysqlCreateAndInsert("tpch.mysql_test_float"));
+        doublePrecisionFloatinPointTests(mysqlDoubleDataType())
+                .execute(getQueryRunner(), mysqlCreateAndInsert("tpch.mysql_test_double"));
+    }
+
+    private static DataTypeTest singlePrecisionFloatingPointTests(DataType<Float> floatType)
+    {
+        // we are not testing Nan/-Infinity/+Infinity as those are not supported by MySQL
```

Comment:
Wow... Does the user get reasonable exception message?

Reply:
I mysql CLI you get: 
![image](https://user-images.githubusercontent.com/890111/83016576-7629bb00-a022-11ea-930d-4213c372fb43.png)

If you try to put NaN as a value via MySQL JDBC you get `java.sql.SQLException: 'NaN' is not a valid numeric or approximate numeric value`